h/api/models.py

```diff
@@ -11,9 +11,45 @@
 from annotator import document
 
 
-def is_file(uri):
-    """Return True if the given uri is a local file uri."""
-    return uri and uri.lower().startswith("file:///")
+def _format_document_link(href, title, link_text, hostname):
+    """Return a document link for the given components.
+
+    Helper function for the .document_link property below.
+
+    The given href, title, link_text and hostname are assumed to be already
+    safely escaped. The returned string will be a Markup object so that
+    it can be rendered in Jinja2 templates without further escaped occurring.
+
+    """
+    if hostname and hostname in link_text:
+        hostname = ""
+
+    def truncate(content, length=50):
+        """Truncate the given string to at most length chars."""
+        if len(content) <= length:
+            return content
+        else:
+            return content[:length] + jinja2.Markup("&hellip;")
+
+    hostname = truncate(hostname)
+    link_text = truncate(link_text)
+
+    if href and hostname:
+        link = ('<a href="{href}" title="{title}">{link_text}</a> '
+                '({hostname})'.format(href=href, title=title,
+                                      link_text=link_text, hostname=hostname))
+    elif hostname and not href:
```

Comment:
The second half of these conditionals isn't needed... it's implied by the fact we're in an `elif` clause and we'd never of got here if it weren't true.